### PR TITLE
Added new method to identify touching conductors.

### PR DESCRIPTION
**Pull request by maxcapodi78**

### CODE REVIEW

File: pyaedt/application/Analysis3D.py [hunk 1]
```diff
@@ -1035,18 +1035,33 @@ def flatten_3d_components(self, component_name=None, purge_history=True, passwor
...
-    def identify_touching_conductors(self):
-        # type: () -> dict
+    def identify_touching_conductors(self, object_name=None):
+        # type: (str) -> dict
         """Identify all the touching components and group in a dictionary. It requires `pyvista`.
 
+        Parameters
+        ----------
+        object_name : str
+            Starting object to check for touching elements. Optional.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1045]:
```suggestion
        object_name : str, optional
            Starting object to check for touching elements. The default is ``None``.
```
Can we describe the default behavior using  this format: The default is ``None``, in which case ....

File: pyaedt/application/Analysis3D.py [hunk 2]
```diff
@@ -1035,18 +1035,33 @@ def flatten_3d_components(self, component_name=None, purge_history=True, passwor
             self.mesh._refresh_mesh_operations()
         return True
 
-    def identify_touching_conductors(self):
-        # type: () -> dict
+    def identify_touching_conductors(self, object_name=None):
+        # type: (str) -> dict
         """Identify all the touching components and group in a dictionary. It requires `pyvista`.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1040]:
```suggestion
        """Identify all touching components and group in a dictionary. This method requires that
        the ``pyvista`` package is installed.
```

**Review by svandenb-dev — APPROVED** [review]

File: pyaedt/application/Analysis3D.py [hunk 3]
```diff
@@ -1034,3 +1034,86 @@ def flatten_3d_components(self, component_name=None, purge_history=True, passwor
...
+
+    def identify_touching_conductors(self, object_name=None):
+        # type: (str) -> dict
+        """Identify all touching components and group in a dictionary. This method requires that
+        the ``pyvista`` package is installed.
+
+        Parameters
+        ----------
+        object_name : str, optional
+            Starting object to check for touching elements. The default is ``None``.
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1046]:
```suggestion
            Starting object to check for touching elements. The default is ``None``.
            
```
Blank line is necessary between sections for proper doc rendering.

File: pyaedt/application/Analysis3D.py [hunk 4]
```diff
@@ -1034,3 +1034,86 @@ def flatten_3d_components(self, component_name=None, purge_history=True, passwor
...
+        Returns
+        -------
+        dict
+
+        """
+        if is_ironpython and settings.aedt_version < "2023.2":  # pragma: no cover
+            self.logger.error("This method requires CPython and Pyvista.")
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1053]:
```suggestion
            self.logger.error("This method requires CPython and PyVista.")
```

File: pyaedt/modeler/cad/object3d.py [hunk 5]
```diff
@@ -296,6 +296,8 @@ def export_image(self, file_path=None):
     @property
     def touching_objects(self):
         """Get the objects that touch one of the vertex, edge midpoint or face of the object."""
```
Comment (PipKat) [inline pyaedt/modeler/cad/object3d.py:298]:
```suggestion
        """Get the objects that touch a vertex, edge midpoint, or face of the object."""
```

File: pyaedt/modeler/cad/object3d.py [hunk 6]
```diff
@@ -865,6 +867,13 @@ def group_name(self, name):
             self._change_property(vgroup)
             self._m_groupName = name
 
+    @property
+    def is_conductor(self):
+        """Check if the object is a conductor or not."""
```
Comment (PipKat) [inline pyaedt/modeler/cad/object3d.py:872]:
```suggestion
        """Check if the object is a conductor."""
```

File: pyaedt/application/Analysis3D.py [hunk 7]
```diff
@@ -1034,3 +1034,87 @@ def flatten_3d_components(self, component_name=None, purge_history=True, passwor
         if not self.design_type == "Icepak":
             self.mesh._refresh_mesh_operations()
         return True
+
+    def identify_touching_conductors(self, object_name=None):
+        # type: (str) -> dict
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis3D.py:1039]: @maxcapodi78 Do we start to add type hints in the methods signature for PyAEDT?

Comment (maxcapodi78) [reply pyaedt/application/Analysis3D.py:1039]: @MaxJPRey I noticed that improves the quality of output documentation and helps in autocomple in IDE

**Pull request merged**